test/java/org/chromium/net/testing/TestUrlRequestCallback.java

```diff
@@ -0,0 +1,389 @@
+package org.chromium.net.testing;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+import static org.chromium.net.testing.CronetTestRule.assertContains;
+
+import android.os.StrictMode;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.chromium.net.CallbackException;
+import org.chromium.net.CronetException;
+import org.chromium.net.InlineExecutionProhibitedException;
+import org.chromium.net.UrlRequest;
+import org.chromium.net.UrlResponseInfo;
+
+/**
+ * Callback that tracks information from different callbacks and and has a method to block thread
+ * until the request completes on another thread. Allows to cancel, block request or throw an
+ * exception from an arbitrary step.
+ */
+public class TestUrlRequestCallback extends UrlRequest.Callback {
```

Comment:
`UrlRequestCallbackTest`?

Reply:
In Cronet, only CronetTestRule has its test. TBH, I suspect this will stay like that...